services/wallet/router/router_v2.go

```diff
@@ -500,8 +505,21 @@ func (r *Router) SuggestedRoutesV2(ctx context.Context, input *RouteInputParams)
 
 	balanceMap, err := r.getBalanceMapForTokenOnChains(ctx, input, selectedFromChains)
 	// return only if there are no balances, otherwise try to resolve the candidates for chains we know the balances for
-	if len(balanceMap) == 0 && err != nil {
-		return nil, errors.CreateErrorResponseFromError(err)
+	if len(balanceMap) == 0 {
+		if err != nil {
+			return nil, errors.CreateErrorResponseFromError(err)
+		}
+	} else {
+		noBalanceOnAnyChain := true
+		for _, value := range balanceMap {
+			if value.Cmp(pathprocessor.ZeroBigIntValue) > 0 {
+				noBalanceOnAnyChain = false
+				break
+			}
+		}
+		if noBalanceOnAnyChain {
+			return nil, ErrNoPositiveBalance
```

Comment:
didn't we break and fix 100 times the possibility of sending with value 0, would this change affect that case?

Reply:
Not sure that I understand this question?

Reply:
Oh I mean, won't this cause an error when we try to do a valid 0-value send?

Reply:
No, it's just for checking the balance on the account for a chain.